ste/sender-blockBlob.go

```diff
@@ -176,7 +176,16 @@ func (s *blockBlobSenderBase) Cleanup() {
 		//    situation. Deletion has very different semantics then, compared to not deleting.
 		deletionContext, cancelFn := context.WithTimeout(context.Background(), 30*time.Second)
 		defer cancelFn()
-		_, _ = s.destBlockBlobURL.Delete(deletionContext, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
+		if jptm.WasCanceled() {
```

Comment:
The heart of the original problem was that this section of code isn't even reached when there's a cancellation.

Also, the same issue exists (of this section not being even reached) for Files and ADLS Gen 2.

Reply:
Oh! I thought the heart of the issue was the cleanup strategy, and was unaware that epilogue code doesn't get called on a cancel.

Reply:
Fixed up. I used a sync.WaitGroup in order to track currently running transfers and awaited their completion in lifecycle manager. As such, this is a fairly global solution and doesn't require much per-subcommand modification.